# Project: Amazon E-Commerce Product Performance & Pricing Analysis

### 🎯 Objective
An analysis of Amazon product data to understand how pricing strategies, discount depths, and review volumes impact customer satisfaction (ratings) and product discoverability.

### 🚀 Product Insights (TL;DR)
* **The Rating Baseline:** High-volume products maintain a stable rating baseline around 4.0 - 4.5⭐, while low-volume items skew heavily.
* **Discount Elasticity:** High discount percentages (like the 64% and 90% seen below) directly correlate with massive review volumes (24k+ reviews), showing promotions drive acquisition.



In [ ]:
import pandas as pd
df = pd.read_csv("amazon.csv")
df.head()
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,₹399,"₹1,099",64%,4.2,"24,269",High Compatibility : Compatible With iPhone 12...,"AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...","Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...","R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...","Satisfied,Charging is really fast,Value for mo...",Looks durable Charging is fine tooNo complains...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,₹199,₹349,43%,4.0,"43,994","Compatible with all Type C enabled devices, be...","AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...","ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...","RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...","A Good Braided Cable for Your Type C Device,Go...",I ordered this cable to connect my phone to An...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Ambrane-Unbreakable-Char...
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,₹199,"₹1,899",90%,3.9,"7,928",【 Fast Charger& Data Sync】-With built-in safet...,"AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...","Kunal,Himanshu,viswanath,sai niharka,saqib mal...","R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...","Good speed for earlier versions,Good Product,W...","Not quite durable and sturdy,https://m.media-a...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Sounce-iPhone-Charging-C...
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,₹329,₹699,53%,4.2,"94,363",The boAt Deuce USB 300 2 in 1 cable is compati...,"AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...","Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...","R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...","Good product,Good one,Nice,Really nice product...","Good product,long wire,Charges good,Nice,I bou...",https://m.media-amazon.com/images/I/41V5FtEWPk...,https://www.amazon.in/Deuce-300-Resistant-Tang...
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,₹154,₹399,61%,4.2,"16,905",[CHARGE & SYNC FUNCTION]- This cable comes wit...,"AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...","rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...","R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...","As good as original,Decent,Good one for second...","Bought this instead of original apple, does th...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Portronics-Konnect-POR-1...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...,₹379,₹919,59%,4,"1,090",SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...,"AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...","Prabha ds,Raghuram bk,Real Deal,Amazon Custome...","R3G3XFHPBFF0E8,R3C0BZCD32EIGW,R2EBVBCN9QPD9R,R...","Received the product without spanner,Excellent...","I received product without spanner,Excellent p...",https://m.media-amazon.com/images/I/41fDdRtjfx...,https://www.amazon.in/Noir-Aqua-Spanner-Purifi...
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,"₹2,280","₹3,045",25%,4.1,"4,118","230 Volts, 400 watts, 1 Year","AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...","Manu Bhai,Naveenpittu,Evatira Sangma,JAGANNADH...","R3DDL2UPKQ2CK9,R2SYYU1OATVIU5,R1VM993161IYRW,R...","ok,everything was good couldn't return bcoz I ...","ok,got everything as mentioned but the me

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df.shape

(1465, 16)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

In [ ]:
df['discounted_price'] = df['discounted_price'].str.replace('₹', '', regex=False).str.replace(',', '', regex=False).astype(float)

df['actual_price'] = df['actual_price'].str.replace('₹', '', regex=False).str.replace(',', '', regex=False).astype(float)


In [ ]:
df['discount_percentage'].dtype

dtype('O')

In [ ]:
df['discount_percentage'] = df['discount_percentage'].astype(str)


In [ ]:
df['discount_percentage'] = df['discount_percentage'].str.replace('%', '', regex=False)
df['discount_percentage'] = pd.to_numeric(df['discount_percentage'], errors='coerce')


In [ ]:
df['rating_count'] = df['rating_count'].str.replace(',', '', regex=False)
df['rating_count'] = pd.to_numeric(df['rating_count'], errors='coerce')



In [ ]:
df['discount_percentage'].head()

,discount_percentage
0,64
1,43
2,90
3,53
4,61


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1465 non-null   object 
 1   product_name         1465 non-null   object 
 2   category             1465 non-null   object 
 3   discounted_price     1465 non-null   float64
 4   actual_price         1465 non-null   float64
 5   discount_percentage  1465 non-null   int64  
 6   rating               1465 non-null   object 
 7   rating_count         1463 non-null   float64
 8   about_product        1465 non-null   object 
 9   user_id              1465 non-null   object 
 10  user_name            1465 non-null   object 
 11  review_id            1465 non-null   object 
 12  review_title         1465 non-null   object 
 13  review_content       1465 non-null   object 
 14  img_link             1465 non-null   object 
 15  product_link         1465 non-null   o

In [ ]:
df['rating'].head()

,rating
0,4.2
1,4.0
2,3.9
3,4.2
4,4.2


In [ ]:
df['rating'].unique()


array(['4.2', '4.0', '3.9', '4.1', '4.3', '4.4', '4.5', '3.7', '3.3',
       '3.6', '3.4', '3.8', '3.5', '4.6', '3.2', '5.0', '4.7', '3.0',
       '2.8', '4', '3.1', '4.8', '2.3', '|', '2', '3', '2.6', '2.9'],
      dtype=object)

In [ ]:
df['rating'] = pd.to_numeric(df['rating'], errors='coerce')


In [ ]:
df['rating'].isna().sum()



np.int64(1)

In [ ]:
df = df.dropna(subset=['rating'])


In [ ]:
df['rating'].isna().sum()

np.int64(0)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1464 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1464 non-null   object 
 1   product_name         1464 non-null   object 
 2   category             1464 non-null   object 
 3   discounted_price     1464 non-null   float64
 4   actual_price         1464 non-null   float64
 5   discount_percentage  1464 non-null   int64  
 6   rating               1464 non-null   float64
 7   rating_count         1462 non-null   float64
 8   about_product        1464 non-null   object 
 9   user_id              1464 non-null   object 
 10  user_name            1464 non-null   object 
 11  review_id            1464 non-null   object 
 12  review_title         1464 non-null   object 
 13  review_content       1464 non-null   object 
 14  img_link             1464 non-null   object 
 15  product_link         1464 non-null   object

In [ ]:
df['discounted_price'].mean()

np.float64(3126.0119057377046)

In [ ]:
df['actual_price'].mean()

np.float64(5447.002923497268)

In [ ]:
df['discount_percentage'].mean()

np.float64(47.71311475409836)

In [ ]:
df['rating'].mean()

np.float64(4.096584699453552)

In [ ]:
avg_discount_percentage = df.groupby('category')['discount_percentage'].mean()
print(avg_discount_percentage)


category
Car&Motorbike|CarAccessories|InteriorAccessories|AirPurifiers&Ionizers                                                    42.0
Computers&Accessories|Accessories&Peripherals|Adapters|USBtoUSBAdapters                                                   78.5
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCHeadsets                                           35.0
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCMicrophones                                        56.5
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCSpeakers                                           46.5
                                                                                                                          ... 
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|GelInkRollerballPens         0.0
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|LiquidIn

In [ ]:
df.sort_values(by='discount_percentage', ascending=False).head(10)

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
695,B097C564GC,"rts [2 Pack] Mini USB C Type C Adapter Plug, T...",Computers&Accessories|Accessories&Peripherals|...,294.0,4999.0,94,4.3,4426.0,【WIDER COMPATIBILITY】This USB C Female to USB ...,"AGO5SRT3ESLNL5WTRQFD5BSIYPQA,AGGNODUMRADB3PQQT...","Naga Sai Gadagottu,Neha Sharma,Amazon Customer...","R3CUNCZTU43JPP,RSO46BN8S4OSU,R2UD5D7T4DZRE5,R2...","Tiny helping hand,Good product,As a whole good...",This eliminates the need to buy chargers for t...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/rts-Adapter-Charging-Con...
334,B0BF57RN3K,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics|WearableTechnology|SmartWatches,1799.0,19999.0,91,4.2,13937.0,Fire-Boltt is India' No 1 Wearable Watch Brand...,"AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...","venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...","R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...","Worth the money,Good smart watch,the screen do...","I liked it, worth the money, light weight and ...",https://m.media-amazon.com/images/I/41WCgGbvwh...,https://www.amazon.in/Fire-Boltt-Bluetooth-Cal...
368,B0BF563HB4,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics|WearableTechnology|SmartWatches,1799.0,19999.0,91,4.2,13937.0,Fire-Boltt is India' No 1 Wearable Watch Brand...,"AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...","venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...","R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...","Worth the money,Good smart watch,the screen do...","I liked it, worth the money, light weight and ...",https://m.media-amazon.com/images/I/41ivjqdXb0...,https://www.amazon.in/Fire-Boltt-Bluetooth-Cal...
364,B0BF54972T,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics|WearableTechnology|SmartWatches,1799.0,19999.0,91,4.2,13937.0,Fire-Boltt is India' No 1 Wearable Watch Brand...,"AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...","venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...","R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...","Worth the money,Good smart watch,the screen do...","I liked it, worth the money, light weight and ...",https://m.media-amazon.com/images/I/41mzbWC6Ak...,https://www.amazon.in/Fire-Boltt-Bluetooth-Cal...
372,B0BF4YBLPX,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics|WearableTechnology|SmartWatches,1799.0,19999.0,91,4.2,13937.0,Fire-Boltt is India' No 1 Wearable Watch Brand...,"AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...","venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...","R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...","Worth the money,Good smart watch,the screen do...","I liked it, worth the money, light weight and ...",https://m.media-amazon.com/images/I/410VGCE+q2...,https://www.amazon.in/Fire-Boltt-Bluetooth-Cal...
380,B0BF54LXW6,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics|WearableTechnology|SmartWatches,1799.0,19999.0,91,4.2,13937.0,Fire-Boltt is India' No 1 Wearable Watch Brand...,"AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...","venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...","R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...","Worth the money,Good smart watch,the screen do...","I liked it, worth the money, light weight and ...",https://m.media-amazon.com/images/I/41Fq27ZjJf...,https://www.amazon.in/Fire-Boltt-Bluetooth-Cal...
602,B08W56G1K9,LAPSTER Spiral Charger Spiral Charger Cable Pr...,Computers&Accessories|Accessories&Peripherals|...,99.0,999.0,90,4.1,8751.0,SPIRAL TUBE CHARGING CABLE PROTECTOR are desig...,"AFNGZSZUISNZ2SMAN3L3OALQXS2Q,AEL2T4V2QAHYRIJ6B...","Happy customer,Screcko,Alka Agrawal,madhu,Udit...","R8UDGYG74HT52,R1ZKTL2UFMHHOQ,R2XQ7ANJA4VF12,RN...","Value for money but,functional,Good,Very easy ...",Ordered this product for charging cables. The ...,https://m.media-amazon.com/images/W/WEBP_402

In [ ]:
df['category'].value_counts()

,count
category,
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|Cables|USBCables,233
Electronics|WearableTechnology|SmartWatches,76
Electronics|Mobiles&Accessories|Smartphones&BasicMobiles|Smartphones,68
"Electronics|HomeTheater,TV&Video|Televisions|SmartTelevisions",63
"Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear",52
...,...
Home&Kitchen|Kitchen&HomeAppliances|SmallKitchenAppliances|RotiMakers,1
"Home&Kitchen|Heating,Cooling&AirQuality|Parts&Accessories|FanParts&Accessories",1
Home&Kitchen|Kitchen&HomeAppliances|SmallKitchenAppliances|StandMixers,1


In [ ]:
df.groupby('category')

In [ ]:
avg_discount = df.groupby('category')['discount_percentage'].mean()
print(avg_discount)

category
Car&Motorbike|CarAccessories|InteriorAccessories|AirPurifiers&Ionizers                                                    42.0
Computers&Accessories|Accessories&Peripherals|Adapters|USBtoUSBAdapters                                                   78.5
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCHeadsets                                           35.0
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCMicrophones                                        56.5
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCSpeakers                                           46.5
                                                                                                                          ... 
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|GelInkRollerballPens         0.0
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|LiquidIn

In [ ]:
avg_discount.sort_values(ascending=False)

,discount_percentage
category,
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|CableConnectionProtectors,90.0
"Electronics|Headphones,Earbuds&Accessories|Earpads",90.0
Electronics|Mobiles&Accessories|MobileAccessories|Décor|PhoneCharms,90.0
"Electronics|Headphones,Earbuds&Accessories|Adapters",88.0
"Computers&Accessories|Accessories&Peripherals|Keyboards,Mice&InputDevices|Keyboard&MiceAccessories|DustCovers",87.5
...,...
OfficeProducts|OfficeElectronics|Calculators|Financial&Business,0.0
OfficeProducts|OfficePaperProducts|Paper|Copy&PrintingPaper|ColouredPaper,0.0
"OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|FountainPens",0.0


In [ ]:
avg_discount.sort_values(ascending=False).head(22)

,discount_percentage
category,
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|CableConnectionProtectors,90.000000
"Electronics|Headphones,Earbuds&Accessories|Earpads",90.000000
Electronics|Mobiles&Accessories|MobileAccessories|Décor|PhoneCharms,90.000000
"Electronics|Headphones,Earbuds&Accessories|Adapters",88.000000
"Computers&Accessories|Accessories&Peripherals|Keyboards,Mice&InputDevices|Keyboard&MiceAccessories|DustCovers",87.500000
Electronics|Mobiles&Accessories|MobileAccessories|Mounts|Shower&WallMounts,82.000000
Computers&Accessories|Components|InternalHardDrives,80.000000
Computers&Accessories|Accessories&Peripherals|Adapters|USBtoUSBAdapters,78.500000
Electronics|Mobiles&Accessories|MobileAccessories|Stands,76.400000


In [ ]:
avg_rating = df.groupby('category')['rating'].mean()
print(avg_rating)

category
Car&Motorbike|CarAccessories|InteriorAccessories|AirPurifiers&Ionizers                                                    3.800000
Computers&Accessories|Accessories&Peripherals|Adapters|USBtoUSBAdapters                                                   4.150000
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCHeadsets                                           3.500000
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCMicrophones                                        3.600000
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCSpeakers                                           4.050000
                                                                                                                            ...   
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSupplies|Pens&Refills|GelInkRollerballPens        4.250000
OfficeProducts|OfficePaperProducts|Paper|Stationery|Pens,Pencils&WritingSu

In [ ]:
avg_rating.sort_values(ascending=False)

,rating
category,
Computers&Accessories|Tablets,4.6
Computers&Accessories|NetworkingDevices|NetworkAdapters|PowerLANAdapters,4.5
Electronics|Cameras&Photography|Accessories|Film,4.5
Computers&Accessories|Components|Memory,4.5
Electronics|HomeAudio|MediaStreamingDevices|StreamingClients,4.5
...,...
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCMicrophones,3.6
"Electronics|HomeTheater,TV&Video|Accessories|3DGlasses",3.5
Computers&Accessories|Accessories&Peripherals|Audio&VideoAccessories|PCHeadsets,3.5


In [ ]:
avg_rating.sort_values(ascending=False).head(5)

,rating
category,
Computers&Accessories|Tablets,4.6
Computers&Accessories|NetworkingDevices|NetworkAdapters|PowerLANAdapters,4.5
Electronics|Cameras&Photography|Accessories|Film,4.5
Computers&Accessories|Components|Memory,4.5
Electronics|HomeAudio|MediaStreamingDevices|StreamingClients,4.5


In [ ]:
df[df['category'] == 'Computers&Accessories|Tablets'].shape[0]

1

In [ ]:
df['category'].value_counts()['Computers&Accessories|Tablets']

np.int64(1)

In [ ]:
category_counts = df['category'].value_counts()
print(category_counts)

category
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|Cables|USBCables                                          233
Electronics|WearableTechnology|SmartWatches                                                                                 76
Electronics|Mobiles&Accessories|Smartphones&BasicMobiles|Smartphones                                                        68
Electronics|HomeTheater,TV&Video|Televisions|SmartTelevisions                                                               63
Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear                                                                52
                                                                                                                          ... 
Home&Kitchen|Kitchen&HomeAppliances|SmallKitchenAppliances|RotiMakers                                                        1
Home&Kitchen|Heating,Cooling&AirQuality|Parts&Accessories|FanParts&Accessories                        

In [ ]:
df['category']

,category
0,Computers&Accessories|Accessories&Peripherals|...
1,Computers&Accessories|Accessories&Peripherals|...
2,Computers&Accessories|Accessories&Peripherals|...
3,Computers&Accessories|Accessories&Peripherals|...
4,Computers&Accessories|Accessories&Peripherals|...
...,...
1460,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...
1461,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...
1462,"Home&Kitchen|Heating,Cooling&AirQuality|RoomHe..."
1463,"Home&Kitchen|Heating,Cooling&AirQuality|Fans|E..."


In [ ]:
df.describe()

,discounted_price,actual_price,discount_percentage,rating,rating_count
count,1464.000000,1464.000000,1464.000000,1464.000000,1462.000000
mean,3126.011906,5447.002923,47.713115,4.096585,18307.376881
std,6946.625442,10878.270001,21.627422,0.291674,42766.096572
min,39.000000,39.000000,0.000000,2.000000,2.000000
25%,323.750000,800.000000,32.000000,4.000000,1191.500000
50%,799.000000,1650.000000,50.000000,4.100000,5179.000000
75%,1999.000000,4303.750000,63.000000,4.300000,17342.250000
max,77990.000000,139900.000000,94.000000,5.000000,426973.000000


In [ ]:
# Check average rating and review volume by discount tiers
df['discount_tier'] = pd.cut(df['discount_percentage'], bins=[0, 30, 60, 100], labels=['Low (<30%)', 'Medium (30-60%)', 'High (>60%)'])
df.groupby('discount_tier', observed=False).agg({
    'rating': 'mean',
    'rating_count': 'mean',
    'product_id': 'count'
}).rename(columns={'product_id': 'total_products'})


/tmp/ipykernel_13067/3473546233.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['discount_tier'] = pd.cut(df['discount_percentage'], bins=[0, 30, 60, 100], labels=['Low (<30%)', 'Medium (30-60%)', 'High (>60%)'])


,rating,rating_count,total_products
discount_tier,,,
Low (<30%),4.150680,20829.564626,294
Medium (30-60%),4.084751,15867.126100,682
High (>60%),4.062642,21444.741419,439


### 📊 Insight: The Discount Paradox
* **Observation:** Products with a **High Discount (>60%)** bring the highest customer traffic, averaging **21,444 reviews** per product.
* **Product Impact:** Deep discounting is highly effective for aggressive user acquisition. However, the average customer rating drops from `4.15` to `4.06`.
* **Recommendation:** While promotions clear inventory and drive traffic, high-discount products need monitoring as they correlate with a slight drop in customer satisfaction.

In [ ]:
# Flag popular items with poor customer experience
poor_experience = df[(df['rating'] < 3.8) & (df['rating_count'] > 5000)]
poor_experience[['product_name', 'category', 'rating', 'rating_count', 'discount_percentage']].head()

,product_name,category,rating,rating_count,discount_percentage
21,tizum HDMI to VGA Adapter Cable 1080P for Proj...,"Electronics|HomeTheater,TV&Video|Accessories|C...",3.7,10962.0,44
45,Wecool Nylon Braided Multifunction Fast Chargi...,Computers&Accessories|Accessories&Peripherals|...,3.3,9792.0,67
90,"Gizga Essentials USB WiFi Adapter for PC, 150 ...",Computers&Accessories|NetworkingDevices|Networ...,3.6,10134.0,66
408,"PTron Boom Ultima 4D Dual Driver, in-Ear Gamin...","Electronics|Headphones,Earbuds&Accessories|Hea...",3.6,18202.0,84
415,PTron Tangent Lite Bluetooth 5.0 Earphones wit...,"Electronics|Headphones,Earbuds&Accessories|Hea...",3.5,83996.0,67


### ⚠️ Product Health Alert: High-Volume Defects
* **Observation:** Multiple heavily discounted products—like the **PTron Tangent Lite Earphones (67% discount)**—maintain massive platform visibility (**83,996 reviews**) despite a low customer satisfaction rating of **3.5⭐**.
* **Product Impact:** Keeping low-quality items pinned to the top via discounts hurts long-term platform trust.
* **Actionable Next Step:** Flag these specific SKUs to the Quality Assurance (QA) and Vendor Management teams for a product health check or supplier review.

In [ ]:
# Save the clean dataset as a CSV file for our next SQL phase
df.to_csv('amazon_clean.csv', index=False)
print("Clean dataset saved successfully!")

Clean dataset saved successfully!
